src/hotspot/share/compiler/compilationPolicy.cpp

```diff
@@ -617,12 +744,12 @@ void CompilationPolicy::handle_counter_overflow(const methodHandle& method) {
 }
 
 // Called with the queue locked and with at least one element
-CompileTask* CompilationPolicy::select_task(CompileQueue* compile_queue) {
+CompileTask* CompilationPolicy::select_task(CompileQueue* compile_queue, JavaThread* THREAD) {
```

Comment:
Not sure why you need `THREAD` here, is it a micro-optimization? Or is it for various `CHECK_*` somewhere? I'd think this should be just `, TRAPS` then?